### PR TITLE
Mysql Only - Add a column at a specific position in the table

### PR DESCRIPTION
**Pull request by alexsante**
This commit provides mysql migrations the ability to add a column before or after another column.
Feature Request: [https://github.com/gobuffalo/pop/issues/3](url)

Example (add `request_id` before the `created_at` column):
```
add_column("request_id","events","uuid", {"null": true, "before":"created_at"})
```

### CODE REVIEW

**Comment by alexsante** [comment]
@markbates any ideas on how I can test this with `buffalo db migrate`?  Would I have to recompile the buffalo binary with the forked package included to test?

**Comment by markbates** [comment]
Yep, that's how you do it. Re-compile the buffalo binary and it will recompile with whatever is in your go path, so if you're in your fork, it should be good. I'll review this today.

**Comment by markbates** [comment]
SQLite is still supported by Pop, please don't remove any SQLite tests.

File: test.sh [hunk 1]
```diff
@@ -26,7 +26,6 @@ function test {
   go test $(go list ./... | grep -v /vendor/)
 }
 
-test "sqlite"
```
Comment (markbates) [inline test.sh]: Please add this back, SQLite is still supported by pop, just off by default. Also, removing this isn't related to this PR. Thanks.

Comment (alexsante) [reply test.sh]: Got it.  I misunderstood.  I've added this line back in.

**Comment by alexsante** [comment]
Hey @markbates, I've added the sqlite test back in.  Sorry about that.

**Comment by markbates** [comment]
When I tried it, I got this error:

```
ERRO[0000] Error: error executing migrations/20180305174620_add_foo_to_widgets.up.fizz, sql: ALTER TABLE widgets ADD COLUMN foo VARCHAR (255) NOT NULL BEFORE created_at;: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'BEFORE created_at' at line 1
```

Here are my migrations:

```
create_table("widgets", func(t) {
	t.Column("id", "uuid", {"primary": true})
	t.Column("name", "string", {})
	t.Column("body", "text", {})
})
```

```
add_column("widgets", "foo", "string", {"before": "created_at"})
```

**Comment by alexsante** [comment]
Good catch @markbates.  For some reason I thought there was a BEFORE option in mysql, but according to their docs, its either `FIRST` or `AFTER`.

```
ADD [COLUMN] col_name column_definition
        [FIRST | AFTER col_name]
```

**Comment by markbates** [comment]
Do you want to add an options for `FIRST`?

**Comment by alexsante** [comment]
@markbates Added `first` option 👍

**Comment by markbates** [comment]
Sorry for the delay in merging this, I've been traveling.

**Pull request merged**